### PR TITLE
Master dig 201

### PR DESCRIPTION
**Pull request by dj1999**

### CODE REVIEW

File: web/modules/custom/diginomica/diginomica.module [hunk 1]
```diff
@@ -152,3 +154,37 @@ function diginomica_node_delete(EntityInterface $entity) {
...
+ *
+ * Add new property for node with name: top_level_category.
+ */
+function diginomica_entity_base_field_info(EntityTypeInterface $entity_type) {
+  if ($entity_type->id() === 'node') {
+    // Add the property just to node.
+    $fields = [];
+    $fields['top_level_category'] = BaseFieldDefinition::create('integer')
+      ->setLabel(t('Top level category'))
+      ->setDescription(t('The term id of top level categori for node.'));
```
Comment (szatom) [inline web/modules/custom/diginomica/diginomica.module]: typo: category

File: web/modules/custom/diginomica/src/DiginomicaGeneral.php [hunk 2]
```diff
@@ -64,7 +68,8 @@ public function checkTermsByRoute(RouteMatchInterface $route_match): bool {
   /**
    * {@inheritdoc}
    */
-  public function getNodeCategoryRootParentTerms(NodeInterface $node) {
+  public function getNodeCategoryRootParentTerms(NodeInterface $node,
+    string $field_name = 'field_category') {
```
Comment (szatom) [inline web/modules/custom/diginomica/src/DiginomicaGeneral.php]: I think it should be field_primary_category

File: web/modules/custom/diginomica/src/DiginomicaGeneral.php [hunk 3]
```diff
@@ -64,7 +68,8 @@ public function checkTermsByRoute(RouteMatchInterface $route_match): bool {
...
     $root_parents = NULL;
     $tids = \array_column($node->get('field_category')->getValue(), 'target_id');
```
Comment (szatom) [inline web/modules/custom/diginomica/src/DiginomicaGeneral.php]: - We should check $field_name 
- We should check if this field exist

File: web/modules/custom/diginomica/src/DiginomicaGeneral.php (hunk 2 above)
Comment (dj1999) [reply web/modules/custom/diginomica/src/DiginomicaGeneral.php]: No this function return the top level parent category term. This is just a helper function in service. But if find that unused this variable fixed it.

**Pull request merged**